subprojects/docs/src/docs/userguide/gradleWrapper.adoc

```diff
@@ -15,106 +15,204 @@
+|`--distribution-type`              |The Gradle distribution type used for the Wrapper. Available options are `bin` and `all`. The default value is `bin`.
+|`--gradle-distribution-url`        |The full URL pointing to Gradle distribution ZIP file. Using this option makes `--gradle-version` and `--distribution-type` obsolete as the URL already contains this information. This option is extremely valuable if you want to host the Gradle distribution inside your company’s network.
+|`--gradle-distribution-sha256-sum` |The SHA256 hash sum used for <<sec:verification,verifying the downloaded Gradle distribution>>.
+|===
+
+Let’s assume the following use case to illustrate the use of the command line options. You would like to generate the Wrapper with version 4.0 and use the `-all` distribution to enable your IDE to enable code-completion and being able to navigate to the Gradle source code. Those requirements are captured by the following command line execution:
 
 ++++
-<sample id="wrapperSimple" dir="userguide/wrapper/simple" title="Wrapper task">
-    <sourcefile file="build.gradle"/>
+<sample id="wrapperCommandLine" dir="userguide/wrapper/simple" title="Providing options to Wrapper task">
+    <output args="wrapper --gradle-version 4.0 --distribution-type all"/>
 </sample>
 ++++
 
-After such an execution you find the following new or updated files in your project directory (in case the default configuration of the Wrapper task is used).
+As a result you can find the desired information in the Wrapper properties file.
+
+.The generated distribution URL
+====
+[source,properties]
+.`gradle/wrapper/gradle-wrapper.properties`
+----
+distributionUrl=https\://services.gradle.org/distributions/gradle-4.0-all.zip
+----
+====
+
+[[sec:upgrading_wrapper]]
+=== Upgrading an existing Wrapper
+
+Projects will typically want to keep up with the times and upgrade their Gradle version to benefit from new features and improvements. One way to upgrade the Gradle version is manually change the `distributionUrl` property in the Wrapper property file. The better and recommended option is to run the `wrapper` task and provide the target Gradle version as described in <<sec:adding_wrapper>>. Using the `wrapper` task ensures that any optimizations made to the Wrapper shell script or batch file with that specific Gradle version are applied to the project. As usual you’d commit the changes to the Wrapper files to version control.
+
+The following console outputs demonstrate the Wrapper upgrade process from Gradle version 4.0 to 4.2.1.
 
 ++++
-<sample id="wrapperSimple" dir="userguide/wrapper/simple" title="Wrapper generated files">
-    <layout after="wrapper">
-        gradlew
-        gradlew.bat
-        gradle/wrapper/gradle-wrapper.jar
-        gradle/wrapper/gradle-wrapper.properties
-    </layout>
+<sample xmlns:xi="http://www.w3.org/2001/XInclude" dir="userguide/wrapper" id="wrapperGradleVersionBeforeUpgrade" title="Checking the version of the existing Wrapper">
```

Comment:
I don't think this sample is necessary. Only explain that `./gradlew --version` exists.

Reply:
The important point I wanted to bring across here is that the Gradle version _actually_ changes. It would be interesting to ask readers if they'd find this helpful. Maybe I am just adding too much detail.

Reply:
I think that's addressed well with the subsequent example.